pipelines/batch/src/main/java/org/openmrs/analytics/FhirEtl.java

```diff
@@ -357,11 +352,42 @@ static void runJsonRead(FhirEtlOptions options) throws IOException {
             .apply(FileIO.readMatches());
     files.apply("Read JSON files", ParDo.of(new ReadJsonFilesFn(options)));
 
-    EtlUtils.runPipelineWithTimestamp(pipeline, options);
+    return pipeline;
   }
 
-  public static void main(String[] args)
-      throws CannotProvideCoderException, PropertyVetoException, IOException, SQLException {
+  /**
+   * Pipeline builder for fetching resources from a FHIR server. The mode of the pipeline is decided
+   * based on the given `options`. There are currently four modes in this priority order:
+   *
+   * <p>1) Fetching directly from the HAPI database if `options.isJdbcModeHapi()` is set.
+   *
+   * <p>2) A combination of direct DB and OpenMRS FHIR API access if `options.isJdbcModeEnabled()`.
+   *
+   * <p>3) Reading from input JSON files if `options.getSourceJsonFilePattern()` is set.
+   *
+   * <p>4) Using FHIR search API if none of the above options are set.
+   *
+   * <p>Depending on the options, the created pipeline may write into Parquet files, push resources
+   * to another FHIR server, write into another database, etc.
+   *
+   * @param options the pipeline options to be used.
+   * @return the created Pipeline instance or null if nothing needs to be done.
+   */
+  static Pipeline buildPipeline(FhirEtlOptions options)
+      throws PropertyVetoException, IOException, SQLException {
+    FhirContext fhirContext = FhirContexts.forR4();
+    if (options.isJdbcModeHapi()) {
+      return buildHapiJdbcPipeline(options);
+    } else if (options.isJdbcModeEnabled()) {
+      return buildOpenmrsJdbcPipeline(options, fhirContext);
+    } else if (!options.getSourceJsonFilePattern().isEmpty()) {
```

Comment:
nit: Can we replace this check with `Strings.isNullOrEmpty` function similar to other places.

Reply:
Done.